src/sentry/tasks/post_process.py

```diff
@@ -431,6 +406,42 @@ def update_event_group(event: "Event") -> None:
     event.group.project.set_cached_field_value("organization", event.project.organization)
 
 
+def process_inbox_adds(
+    event: "Event", is_reprocessed: bool, is_new: bool, is_regression: bool
+) -> bool:
+    from sentry.models import GroupInboxReason
+    from sentry.models.groupinbox import add_group_to_inbox
+
+    has_reappeared = not is_new
+
+    if is_reprocessed:
+        with sentry_sdk.start_span(op="tasks.post_process_group.add_group_to_inbox"):
+            try:
+                if is_new:
+                    add_group_to_inbox(event.group, GroupInboxReason.REPROCESSED)
+            except Exception:
+                logger.exception("Failed to add group to inbox for reprocessed groups")
+    else:
+        # we process snoozes before rules as it might create a regression
+        # but not if it's new because you can't immediately snooze a new group
+        try:
+            if has_reappeared:
+                has_reappeared = process_snoozes(event.group)
+        except Exception:
+            logger.exception("Failed to process snoozes for group")
```

Comment:
Ideally we could extract this into its own function and pass `has_reappeared` into `process_inbox_adds`

Reply:
sure thing, i think i can make this work semantically (though we'll still have `process_snooze` have `add_group_to_inbox` logic)